### DISCUSSION TITLE
Logging invalid commands that players enter for fine tuning engine commands and aliases.

### DISCUSSION THREAD

**Discussion by ArmoredArtist** [GoMudEngine/GoMud#78] [Ideas]
I would like to suggest logging invalid commands that players enter. It could be useful to track what commands players are instinctively trying but aren’t supported yet. Over time, you could review those logs to identify new features or quality-of-life command improvements that align with what players naturally expect. This also allows you to collect data (and clues) from any random person who joins and vanishes into the aether never to return again.

It might be useful to have it collect the invalid commands in two files. One with just the invalid commands to look at (cleaner file to view) and another with a small snippit of the game play the user was involved in when he tried the command to give you more context if you need to look up why they used the command. This would be more cluttery of a log file obviously (ctrl f ftw)

Source: I naturally tried to use some commands that did not exist last night when I tested the game. Such as (from fuzzy memory) coins, money, bank, gold when trying to check my finances.

**Comment by Volte6** [comment]
Yeah that's a good idea... in testing with people I try to watch what they type and ask what they expected to happen, and then consider adding it as a permanent alias here: https://github.com/Volte6/GoMud/blob/master/_datafiles/keywords.yaml#L158-L194

For now I've added it in-memory in this PR: https://github.com/Volte6/GoMud/pull/80

There is a new admin command `badcommands` which lists them out like this:

<img width="519" alt="image" src="https://github.com/user-attachments/assets/2f3ebb34-5542-491b-876e-18956006a409">

**Comment by ArmoredArtist** [comment]
I feel like you implement things even faster than I can type the suggestions. Nice work!

**Comment by Volte6** [comment]
Closing as completed
